packages/vite/src/node/plugins/esbuild.ts

```diff
@@ -27,7 +27,7 @@ import { searchForWorkspaceRoot } from '..'
 const debug = createDebugger('vite:esbuild')
 
 const INJECT_HELPERS_IIFE_RE =
-  /(.*)(var [^\s]+=function\([^)]*?\){"use strict";)(.*)/
```

Comment:
This change is interesting, if someone changes the `generatedCode` to `es5` then this would fail. Maybe we could do `(?:const|var)` instead?

Reply:
Oh yes it does happen.
But I wonder if there is a reason to change `generatedCode` to `es5`.
esbuild transpiles the output of rollup code so setting it won't affect the output expect `esbuild: false` is set.
Also it won't work with es5 without transpiling because preload helper uses es2015.

Reply:
That's also true. I guess someone would need a plugin to go down to es5, but I think making it more flexible could be more future proof. And thanks for making the change too!